src/autolabel/models/tgi.py

```diff
@@ -0,0 +1,212 @@
+                                    "logprobs": {
+                                        "top_logprobs": self.parse_logprobs(
+                                            response["details"]["tokens"]
+                                        )
+                                    }
+                                }
+                                if self.config.confidence()
+                                else None
+                            ),
+                        )
+                    ]
+                )
+                errors.append(None)
+                latencies.append(latency)
+            except Exception as e:
+                logger.exception(
+                    f"Unable to generate prediction: {e}",
+                )
+                generations.append([Generation(text="")])
+                errors.append(
+                    LabelingError(
+                        error_type=ErrorType.LLM_PROVIDER_ERROR, error_message=str(e)
+                    )
+                )
+                latencies.append(0)
+        return RefuelLLMResult(
+            generations=generations, errors=errors, latencies=latencies
+        )
+
+    async def _alabel(self, prompts: List[str]) -> RefuelLLMResult:
+        generations = []
+        errors = []
+        latencies = []
+        try:
+            requests = [self._alabel_with_retry(prompt) for prompt in prompts]
+            responses = await asyncio.gather(*requests)
+            for response, latency in responses:
+                response = json.loads(response.text)[0]
+                generations.append(
+                    [
+                        Generation(
+                            text=response["generated_text"],
+                            generation_info=(
+                                {
+                                    "logprobs": {
+                                        "top_logprobs": self.parse_logprobs(
+                                            response["details"]["tokens"]
+                                        )
+                                    }
+                                }
+                                if self.config.confidence()
+                                else None
+                            ),
+                        )
+                    ]
+                )
+                errors.append(None)
+                latencies.append(latency)
+        except Exception as e:
+            logger.exception(
+                f"Unable to generate prediction: {e}",
+            )
+            generations.append([Generation(text="")])
+            errors.append(
+                LabelingError(
+                    error_type=ErrorType.LLM_PROVIDER_ERROR, error_message=str(e)
+                )
+            )
+            latencies.append(0)
+        return RefuelLLMResult(
+            generations=generations, errors=errors, latencies=latencies
+        )
+
+    def parse_logprobs(self, logprobs: List):
```

Comment:
Do we need this parse_logprobs function? The confidence parsing logic should be able to understand the format in which confidence is returned by TGI/Lorax?